python/cudf/cudf/core/_base_index.py

```diff
@@ -1414,6 +1423,128 @@ def from_pandas(cls, index, nan_as_null=None):
     def _constructor_expanddim(self):
         return cudf.MultiIndex
 
+    def drop_duplicates(
+        self, keep="first", nulls_are_equal=True,
+    ):
+        """
+        Drop duplicate rows in index.
+
+        keep : ["first", "last", False]
+            "first" will keep first of duplicate, "last" will keep last of the
+            duplicate and "False" drop all duplicate.
+        nulls_are_equal: bool, default True
+            Null elements are considered equal to other null elements.
+        """
+
+        # This utilizes the fact that all `Index` is also a `Frame`.
+        result = self.__class__._from_columns(
+            drop_duplicates(
+                list(self._columns),
+                keys=range(len(self._data)),
+                keep=keep,
+                nulls_are_equal=nulls_are_equal,
+            ),
+            self._column_names,
+        )
+        result._copy_type_metadata(self)
+        return result
+
+    def dropna(self, how="any"):
+        """
+        Drop null rows from Index.
+
+        how : {"any", "all"}, optional
+            Specifies how to decide whether to drop a row.
+            "any" (default) drops rows containing at least
+            one null value. "all" drops only rows containing
+            *all* null values.
+        """
+
+        # This is to be consistent with IndexedFrame.dropna to handle nans
+        # as nulls by default
```

Comment:
Just checking, is this consistent with pandas?

Reply:
This prompted me to look into dropna index tests. And couldn't find any :(

Reply:
But back to your question - I don't think you can mix nan and null in index, since there isn't a nullable float type. So this process really is only for cudf to drop any nan/null, arguably something expected from a pandas user perspective.

Reply:
Technically, we have a nullable float type in pandas (`pd.Float64Dtype` and `pd.Float32Dtype`). It is hard to construct a nullable float series that contains both nulls and nans, but technically possible:

```python
>>> sr = pd.Series([0,0,0], dtype=pd.Float64Dtype())
>>> nans = sr / sr
>>> nans
0    NaN
1    NaN
2    NaN
dtype: Float64
>>> nans[0] = pd.NA
>>> nans
0    <NA>
1     NaN
2     NaN
dtype: Float64
```
Usually when trying to directly set a `nan` into one of these series, an `<NA>` will end up in the result instead. As for how it works as an index, it seems there might be some bugs:

```python
>>> pd.Series([1,2,3], index=nans)
NaN    1
NaN    2
NaN    3
dtype: int64
>>> pd.Series([1,2,3], index=nans).index
Index([<NA>, nan, nan], dtype='object')
```

Overall it seems like these are new and not fully supported everywhere, and how they will work in a lot of these edge cases is still up in the air. As such I am not sure it is worth spending too much time matching what they do exactly.

Reply:
Not sure how to create a nullable float pandas array yet.
```python
In [21]: pd.Index([1.0, float('nan'), None], pd.Float64Dtype())
Out[21]: Index([1.0, <NA>, <NA>], dtype='object')

In [22]: pd.Series([1.0, float('nan'), None], pd.Float64Dtype())
...
TypeError: 'Float64Dtype' object is not iterable
```

Reply:
But haven't found a way to construct an index with this dtype yet.

Reply:
Ah, apparently github didn't refresh the above comments from @brandon-b-miller . I edited my answers without seeing him already replied.

Reply:
This seems fine to me then. I agree, let's not waste time matching partly supported behaviors in pandas.